### PR TITLE
ipn/ipnlocal: make app connector configuration concurrent

### PR DESCRIPTION
**Pull request by raggi**
If there are routes changes as a side effect of an app connector configuration update, the connector configuration may want to reenter a lock, so must be started asynchronously.

Updates tailscale/corp#16833
Signed-off-by: James Tucker <james@tailscale.com>

### CODE REVIEW

**Review by catzkorn — APPROVED** [review]

File: ipn/ipnlocal/local.go [hunk 1]
```diff
@@ -3460,8 +3460,11 @@ func (b *LocalBackend) reconfigAppConnectorLocked(nm *netmap.NetworkMap, prefs i
 	slices.SortFunc(routes, func(i, j netip.Prefix) int { return i.Addr().Compare(j.Addr()) })
 	domains = slices.Compact(domains)
 	routes = slices.Compact(routes)
-	b.appConnector.UpdateRoutes(routes)
-	b.appConnector.UpdateDomains(domains)
+	// The following updates may want to take preferences changes, so make them concurrent.
+	go func() {
+		b.appConnector.UpdateRoutes(routes)
+		b.appConnector.UpdateDomains(domains)
+	}()
```
Comment (bradfitz) [inline ipn/ipnlocal/local.go:3467]: do you care about scheduling order?

Comment (bradfitz) [reply ipn/ipnlocal/local.go:3467]: (Maybe we should move `type execQueue` out somewhere shared... that type runs func(s) in another goroutine but in order....)

Comment (raggi) [reply ipn/ipnlocal/local.go:3467]: yeah, I think you're right

File: ipn/ipnlocal/local.go [hunk 2]
```diff
@@ -3460,8 +3460,7 @@ func (b *LocalBackend) reconfigAppConnectorLocked(nm *netmap.NetworkMap, prefs i
 	slices.SortFunc(routes, func(i, j netip.Prefix) int { return i.Addr().Compare(j.Addr()) })
 	domains = slices.Compact(domains)
 	routes = slices.Compact(routes)
-	b.appConnector.UpdateRoutes(routes)
-	b.appConnector.UpdateDomains(domains)
+	b.appConnector.UpdateDomainsAndRoutes(domains, routes)
```
Comment (catzkorn) [inline ipn/ipnlocal/local.go:3463]: Nice! And TIL I learn about a new thing 😄

**Review by catzkorn — APPROVED** [review]
Thanks @bradfitz for flagging 😃

**Pull request merged**